docs/en/advanced_config/bootloader_secure_boot.md

```diff
@@ -0,0 +1,323 @@
+# Secure Boot and Firmware Signature Verification
+
+PX4 includes support for secure boot, which prevents unauthorized or tampered firmware from running on the flight controller.
+This feature uses cryptographic signature verification to ensure only trusted firmware can be executed.
+
+::: warning
+Secure boot is an advanced feature primarily intended for commercial applications and OEM manufacturers who need to protect their firmware from tampering.
+Improper configuration can brick your device.
+:::
+
+## Overview
+
+The PX4 secure boot implementation provides:
+
+- **Firmware signature verification**: The bootloader verifies cryptographic signatures before executing firmware.
+- **Tamper protection**: Unsigned or improperly signed firmware will not be executed.
+- **Key-based authentication**: Uses cryptographic keys stored in the device.
+- **Table of Contents (TOC)**: A structured format for organizing firmware components with security metadata.
+
+The secure boot feature is available on NuttX-based flight controllers.
+
+## Enabling Secure Boot
+
+### Prerequisites
+
+Before enabling secure boot, you need:
+
+1. A board configuration that supports secure boot (NuttX-based flight controllers)
+2. Understanding of the build system and firmware signing process
+3. Secure storage for your private keys
+4. A workflow for signing firmware images
+
+### Board Configuration
+
+To enable secure boot on your board:
+
+1. **Enable security in bootloader**: Add to your board's bootloader configuration file (e.g., `boards/[vendor]/[board]/nuttx-config/bootloader/defconfig`):
+
+   ```sh
+   CONFIG_BOOTLOADER_USE_SECURITY=y
+   ```
+
+2. **Set crypto algorithm**: Define the signing algorithm in your board's `hw_config.h`:
+
+   ```c
+   #define BOOTLOADER_SIGNING_ALGORITHM CRYPTO_ED25519
+   ```
+
+3. **Configure key storage**: Implement or configure the keystore backend for your board
+4. **Build bootloader with crypto support**: Ensure `PX4_CRYPTO` is defined in your board's CMake configuration
+
+### Key Management
+
+::: warning CRITICAL
+Keep your private keys secure! If private keys are lost, you cannot sign new firmware. If private keys are compromised, unauthorized firmware can be created.
+:::
+
+Key management involves:
+
+1. **Key generation**: Generate cryptographic key pairs (public and private keys)
+2. **Key storage**:
+   - Private keys: Store securely on your build/signing server (never on the device)
+   - Public keys: Store in the device's keystore (accessible to bootloader)
+3. **Key indices**: Each key has an index (0-15) used to reference it in the TOC
+
+#### Generating ED25519 Keys
```

Comment:
1. Is there any reason to use ED25519 vs other supported algorithms?
2. Is this section correct? Specifically this seems to create a python script to create a key  pair. Is this the best way? Do we have scripts/tools in the tree that do this for you (and if not, why don't we include this script in our source?)